// dicebear/lib/utils/getStyleCommandSchema.js
import { schema as coreSchema } from '@dicebear/core';
export function getStyleCommandSchema(style) {
    var _a;
    return {
        $schema: 'http://json-schema.org/draft-07/schema#',
        type: 'object',
        properties: {
            count: {
                description: 'Defines how many avatars to create.',
                type: 'number',
                default: 1,
            },
            format: {
                type: 'string',
                enum: ['svg', 'png', 'jpg', 'jpeg', 'webp', 'avif', 'json'],
                default: 'svg',
            },
            exif: {
                description: 'Include Exif Metadata',
                type: 'boolean',
                default: false,
            },
            json: {
                description: 'Save JSON file in addition to image file',
                type: 'boolean',
                default: false,
            },
            ...coreSchema.properties,
            ...(_a = style.schema) === null || _a === void 0 ? void 0 : _a.properties,
        },
    };
}
